File: lib/features/progress/data/models/user_stats.dart
import 'package:isar/isar.dart';

part 'user_stats.g.dart';

@collection
class UserStats {
  Id id = 1; // Singleton

  int totalXp = 0;
  
  int currentLevel = 1;
  
  int currentStreak = 0;
  
  int longestStreak = 0;
  
  DateTime? lastActiveDate;
}

File: backend/models/UserStats.js
const mongoose = require('mongoose');

const userStatsSchema = new mongoose.Schema({
  userId: { type: String, ref: 'User', required: true, unique: true },
  totalXp: { type: Number, default: 0 },
  currentLevel: { type: Number, default: 1 },
  currentStreak: { type: Number, default: 0 },
  longestStreak: { type: Number, default: 0 },
  lastActiveDate: { type: Date },
  updatedAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('UserStats', userStatsSchema);
